samcli/lib/init/template_modifiers/application_insights_template_modifier.py

```diff
@@ -0,0 +1,95 @@
+"""
+Class used to parse and update template when application-insights is enabled
+"""
+import logging
+from typing import Any
+from samcli.lib.init.template_modifiers.cli_template_modifier import TemplateModifier
+
+LOG = logging.getLogger(__name__)
+
+
+class ApplicationInsightsTemplateModifier(TemplateModifier):
+    import ruamel.yaml
+    from ruamel.yaml import YAML
+    from ruamel.yaml.comments import CommentedMap
+
+    TYPE_KEY = "Type"
+    RESOURCES_KEY = "Resources"
+    PROPERTIES_KEY = "Properties"
+    NAME_KEY = "Name"
+    RESOURCE_QUERY_KEY = "ResourceQuery"
+    RESOURCE_GROUP_NAME_KEY = "ResourceGroupName"
+    AUTO_CONFIG_ENABLED_KEY = "AutoConfigurationEnabled"
+    DEPENDS_ON_KEY = "DependsOn"
+    RESOURCE_GROUP_TYPE = "AWS::ResourceGroups::Group"
```

Comment:
We usually keep resource definition types in this file, can we move this one and `AWS::ApplicationInsights::Application` there as well?

https://github.com/aws/aws-sam-cli/blob/d0f99f2f0160e53f04f08949ac3f779eb6f61f79/samcli/lib/utils/resources.py